libraries/bezier-rs/src/subpath/solvers.rs

```diff
@@ -59,6 +59,62 @@ impl Subpath {
 
 		intersection_t_values
 	}
+
+	/// Return the min and max corners that represent the bounding box of the subpath.
+	pub fn bounding_box(&self) -> Option<[DVec2; 2]> {
+		let bounding_boxes = self.iter().map(|bezier| bezier.bounding_box()).collect::<Vec<[DVec2; 2]>>();
+		if bounding_boxes.is_empty() {
+			// There is no bounding box for empty subpath.
+			None
+		} else {
+			let mut min_point = bounding_boxes[0][0];
+			let mut max_point = bounding_boxes[0][1];
+			for bounding_box in bounding_boxes {
+				// We take min of mins and max of maxes.
+				min_point = min_point.min(bounding_box[0]);
+				max_point = max_point.max(bounding_box[1]);
+			}
+			Some([min_point, max_point])
+		}
+	}
+
+	/// Returns two lists of `t`-values representing the local extrema of the `x` and `y` parametric subpaths respectively.
+	/// The list of `t`-values returned are filtered such that they fall within the range `[0, 1]`.
+	pub fn local_extrema(&self) -> [Vec<f64>; 2] {
+		let (mut x_extremas, mut y_extremas) = (Vec::new(), Vec::new());
+		let number_of_curves = self.len_segments() as f64;
+
+		for (index, bezier) in self.iter().enumerate() {
+			// Convert t-values of bezier curve to t-values of subpath
+			let extremas = bezier.local_extrema();
+			x_extremas.extend(extremas[0].iter().map(|t| ((index as f64) + t) / number_of_curves).collect::<Vec<f64>>());
+			y_extremas.extend(extremas[1].iter().map(|t| ((index as f64) + t) / number_of_curves).collect::<Vec<f64>>());
+		}
+
+		// TODO: Consider the shared point between adjacent beziers.
+		[x_extremas, y_extremas]
+	}
+
+	/// Returns list of `t`-values representing the inflection points of the subpath.
+	/// The list of `t`-values returned are filtered such that they fall within the range `[0, 1]`.
+	pub fn inflections(&self) -> Vec<f64> {
+		let number_of_curves = self.len_segments() as f64;
+		let inflection_t_values: Vec<f64> = self
+			.iter()
+			.enumerate()
+			.flat_map(|(index, bezier)| {
+				bezier
+					.inflections()
+					.into_iter()
+					// Convert t-values of bezier curve to t-values of subpath
+					.map(|t| ((index as f64) + t) / number_of_curves)
+					.collect::<Vec<f64>>()
```

Comment:
@0HyperCube can this inner `collect` be avoided?

Reply:
Yes, you should be able to just return the iterator I think.

Reply:
I tried just deleting that last line (`.collect...`) but it didn't want to compile.

Reply:
The map statement has to `move` the index so that it is not dropped (but this is fine as usize is copy) - `.map(move |t| ((index as f64) + t) / number_of_curves)`

Reply:
Oh, I briefly tried putting a `move` in but I put it in the wrong place. Thanks.